client/hwlib/src/models/request_validators.rs

```diff
@@ -77,20 +77,25 @@ pub struct CertificationStatusRequest {
 }
 
 impl CertificationStatusRequest {
-    pub fn new(paths: Paths) -> Result<Self> {
+    pub fn new(paths: Paths, runner: Option<&dyn CommandRunner>) -> Result<Self> {
```

Comment:
Passing an `Option` here and then using `unwrap_or` inside of the function as a way of having a default value is a little odd. The API will be easier to understand at the call site if you always require passing the `CommandRunner` impl, or alternatively you could keep `new` with the signature it had before and change it to a wrapper for `new_with_runner`:
```rust
impl CertificationStatusRequest {
    pub fn new(paths: Paths) -> Result<Self> {
        Self::new_with_runner(paths: Paths, runner: &RealCommandRunner)
    }

    pub fn new_with_runner(paths: Paths, runner: &dyn CommandRunner) -> Result<Self> {
        // Current impl of "new" minus the call to unwrap_or
    }
}
```

Reply:
Updated in f497200